### ISSUE TITLE
System clipboard on wayland

### ISSUE THREAD

**Issue by seankhliao** [neovim/neovim#10931]
- `nvim --version`:
```
NVIM v0.4.0-1944-g38806f23e
Build type: RelWithDebInfo
LuaJIT 2.0.5
Compilation: /usr/bin/cc -march=native -O2 -pipe -fstack-protector-strong -fno-plt -O2 -g -DMIN_LOG_LEVEL=3 -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/home/arccy/.cache/yay/neovim-git/src/neovim-git/build/config -I/home/arccy/.cache/yay/neovim-git/src/neovim-git/src -I/usr/include -I/home/arccy/.cache/yay/neovim-git/src/neovim-git/build/src/nvim/auto -I/home/arccy/.cache/yay/neovim-git/src/neovim-git/build/include
Compiled by arccy

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```
- `vim -u DEFAULTS` (version: ) behaves differently?
- Operating system/version:
`archlinux  5.2.11-arch1-1-ARCH`
- Terminal name/version:
`kitty 0.14.4`
- `$TERM`:
`xterm-256color`

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
enter visual mode
highlight text
"+y or "*y or y (with clipboard=unnamedplus)
```

### Actual behaviour
text not synced to clipboard

### Expected behaviour
text synced to clipboard

### other
from `:checkhealth`:
```
health#provider#check
========================================================================
## Clipboard (optional)
  - OK: Clipboard tool found: wl-copy
```

`wl-copy` and `wl-paste` work normally when called directly from terminal
window manager is sway

**Comment by justinmk** [comment]
related (reported as solved): https://github.com/neovim/neovim/issues/9806 .

It's not clear if `wl-copy` + `wl-paste` + sway are enough to reproduce this, since apparently it works for some (most?) users. Perhaps there is a new `wl-copy`/`wl-paste` option/argument needed? Would be great if someone figures this out.

**Comment by seankhliao** [comment]
it used to work for me, I think it might have broke sometime in the last week

**Comment by justinmk** [comment]
so last week you were on last week's master, meaning perhaps the change happened somewhere since  4cedc8444e20b5b9a558c66a96851c39abedef70  ?

**Comment by seankhliao** [comment]
I checked my logs and did some upgrade/downgrade testing and I'm going to blame this on `wl-clipboard` not having a recent release

my guess is `wl-clipboard` needs to be built against the recent `wlroots` 0.7 (shared wayland backend) that was released at the same time time as sway 1.2 (it wasn't an issue with sway 1.2_rc2)

**tldr**: (re)build wl-clipboard from `master` against new releases

sorry for the issue!
